specs/time_series_api.adoc

```diff
@@ -0,0 +1,285 @@
+below.
+
+Sample Request
+
+----------
+GET /monitoring/nodes/<node_name>/monitored_types
+----------
+
+Sample Response
+
+----------
+Status: 200 OK
+
+[{
+  type: string,
+  description: the resource name,
+}]
+
+ex:
+['resource_1', 'resource_2', ...]
+----------
+
+The above response which is an array of resource name strings gives different
+statistics that tendrl's monitoring stack maintains in its time series db.
+
+
+Sample Request
+
+----------
+GET /monitoring/nodes/<node_name>/<resource_name>/stats
+----------
+
+Sample Response
+
+----------
+Status: 200 OK
+[
+  {
+    "target": {
+      type: string,
+      description: full qualified name of the resource including the
+        node name, a default prefix which gives a namespace to the stats in
+        graphite and name of the resource.
+    },
+    "datapoints": [
+      [{
+          type: float,
+          description: Instant value of target
+        }, {
+          type: long,
+          description: Timestamp at which the value was observed
+      }],....
+    ]
+  }
+]
+
+ex: [
+  {
+    "target": "{graphite_prefix.host_name.metric_name}",
+    "datapoints": [
+      [5.50091681946991, 1478495700],....
+    ]
+  }
+]
+----------
+
+The above response gives the statistics for last 24 hours corresponding to the
+'resource_name' passed to it. If a time range is passed as optional arguements
+to the above api, then the api responds with the statistics for that time
+range.
+
+==== Tendrl/common impact:
+
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+None
+
+=== Security impact:
+
+The apis exposed by Tendrl/performance-monitoring although are only getters,
+need to be covered by a layer of authentication.It could even be as sample as
+a shared token by which its access is enabled only to Tendrl/tendrl-api.
+
+=== Other end user impact:
+
+None
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+Some of the config options as below need to be made available:
+
+----
+[time_series]
+time_series_db = graphite
+time_series_db_server = 0.0.0.0
+time_series_db_port = 80
+
+[tendrl_performance]
+performance_server_interface = 0.0.0.0
+
+
+where the fields mean the following:
+
+* time_series_db - to indicate the currently used time-series db so that the
+                   corresponding plugin can be used
+* time_series_db_server - ip of the server where the time-series db resides.
+* time_series_db_port - port on which time series db can be acceseed for read.
+* performance_server_interface - the network interface only on which the
+                                 Tendrl/performance_monitoring serves
+----
+
+=== Developer impact:
+
+This has extensive impact on 2 modules tendrl-api and performance-monitoring
+and hence warrants a very high collaboration between the owners of the
+2 modules
+
+== Implementation:
+
+
+=== Assignee(s):
+
+
+Who is leading the writing of the code? Or is this a spec where you're
```

Comment:
I feel line 253-257 is not required. Its retained from sample I feel. Please remove.